tests/testsuite/doc.rs

```diff
@@ -777,6 +777,36 @@ fn doc_target() {
         .is_file());
 }
 
+#[cargo_test]
+fn doc_target_and_open() {
+    const TARGET: &str = "i686-unknown-linux-gnu";
+
+    let p = project()
+        .file(
+            "src/lib.rs",
+            "
+            /// test
+            pub fn foo() {}
+            ",
+        )
+        .build();
+
+    p.cargo("doc --verbose --open --target")
+        .arg(TARGET)
```

Comment:
For the target here, I think it would be fine to use `rustc_host()` from `cargo_test_support`.